ownCloud/Settings/SecuritySettingsSection.swift

```diff
@@ -0,0 +1,177 @@
+
+    func toString() -> String {
+        switch self {
+        case .always:
+            return "Always".localized
+        case .oneMinute:
+            return "1 minute".localized
+        case .fiveMinutes:
+            return "5 minutes".localized
+        case .thirtyMinutes:
+            return "30 minutes".localized
+        }
+    }
+}
+
+class SecuritySettingsSection: SettingsSection {
+
+    var frequency: SecurityAskFrequency {
+        willSet {
+            self.userDefaults.set(newValue.rawValue, forKey: SecuritySettingsFrequencyKey)
+        }
+    }
+
+    var isPasscodeSecurityEnabled: Bool {
+        willSet {
+            self.userDefaults.set(newValue, forKey: SecuritySettingsPasscodeKey)
+        }
+
+    }
+    var isBiometricalSecurityEnabled: Bool {
+        willSet {
+            self.userDefaults.set(newValue, forKey: SecuritySettingsBiometricalKey)
+        }
+    }
+
+    // MARK: - Upload Settings Cells
+
+    private var frequencyRow: StaticTableViewRow?
+    private var passcodeRow: StaticTableViewRow?
+    private var biometricalRow: StaticTableViewRow?
+
+    override init(userDefaults: UserDefaults) {
+
+        frequency = SecurityAskFrequency.init(rawValue: userDefaults.integer(forKey: SecuritySettingsFrequencyKey))!
+        isPasscodeSecurityEnabled = userDefaults.bool(forKey: SecuritySettingsPasscodeKey)
+        isBiometricalSecurityEnabled = userDefaults.bool(forKey: SecuritySettingsBiometricalKey)
+
+        super.init(userDefaults: userDefaults)
+
+        self.headerTitle = "Security".localized
+        self.identifier = SecuritySectionIdentifier
+
+        createRows()
+        updateUI()
+    }
+
+    // MARK: - Creation of the rows.
+    func createRows() {
+
+        // Creation of the frequency row.
+        frequencyRow = StaticTableViewRow(subtitleRowWithAction: { (row, _) in
+            if let vc = self.viewController {
+
+                let newVC = StaticTableViewController(style: .grouped)
+                let frequencySection = StaticTableViewSection(headerTitle: "Select the frequency that security should be showed".localized, footerTitle: nil)
+
+                var radioButtons: [[String : Any]] = []
+
+                for frequency in SecurityAskFrequency.all {
+                    radioButtons.append([frequency.toString() : frequency.rawValue])
+                }
+
+                frequencySection.add(radioGroupWithArrayOfLabelValueDictionaries: radioButtons, radioAction: { (row, _) in
+                    if let rawFrequency = row.value! as? Int, let frequency = SecurityAskFrequency.init(rawValue: rawFrequency) {
+                        self.frequency = frequency
+                        self.frequencyRow?.cell?.detailTextLabel?.text = frequency.toString()
+                    }
+                }, groupIdentifier: "frequency-group-identifier", selectedValue: SecurityAskFrequency.always.rawValue, animated: true)
+
```

Comment:
If I pick a duration for the passcode lock (i.e. 30 mins) and go back to the settings screen, it shows the 30 mins. But if I tap on that entry again, the selection defaults back to "Always".

You can either use `func setSelected(_ value: Any, groupIdentifier: String)`to change the value after creation - or pass in the current setting as `selectedValue` during creation.